src/core/modules/StarMgr.cpp

```diff
@@ -916,9 +916,14 @@ void StarMgr::loadCultureSpecificNames(const QJsonObject& data, const QMap<QStri
 	for (auto it = data.begin(); it != data.end(); ++it)
 	{
 		const auto key = it.key();
+		// Let's allow Hipparcos and Gaia designations only
 		if (key.startsWith("HIP "))
 		{
-			loadCultureSpecificNameForHIP(it.value().toArray(), key.mid(4).toULongLong());
+			loadCultureSpecificNameForStar(it.value().toArray(), key.mid(4).toULongLong());
```

Comment:
Now it should be `toUInt()` here.

Reply:
Why? The second parameter for method `loadCultureSpecificNameForStar` has _StarId_ type

Reply:
Because we are parsing a HIP, which cannot exceed a 32-bit int range.

Reply:
I still see no reasons to twice type conversion

Reply:
It's not done twice. It parses the string once (also validating the range), then zero-extends once.